File: php/features.php
<?php
declare(strict_types=1);

// IpGeolocationApi3 SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class IpGeolocationApi3Features
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new IpGeolocationApi3BaseFeature();
            case "test":
                return new IpGeolocationApi3TestFeature();
            default:
                return new IpGeolocationApi3BaseFeature();
        }
    }
}

File: py/ipgeolocationapi3_sdk/features.py
# IpGeolocationApi3 SDK feature factory

from ipgeolocationapi3_sdk.feature.base_feature import IpGeolocationApi3BaseFeature
from ipgeolocationapi3_sdk.feature.test_feature import IpGeolocationApi3TestFeature


def _make_feature(name):
    features = {
        "base": lambda: IpGeolocationApi3BaseFeature(),
        "test": lambda: IpGeolocationApi3TestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
